operators/pkg/controller/elasticsearch/driver/upgrade_test.go

```diff
@@ -0,0 +1,321 @@
+		wantNewPartition map[string]int32
+		wantSyncedFlush  bool
+	}{
+		{
+			name: "single sset upgrade",
+			args: args{
+				statefulSets: sset.StatefulSetList{
+					nodespec.TestSset{
+						Name:      "default",
+						Replicas:  1,
+						Master:    true,
+						Data:      true,
+						Partition: 1,
+						Status: appsv1.StatefulSetStatus{
+							CurrentRevision: "a",
+							UpdateRevision:  "b",
+						},
+					}.Build(),
+				},
+				esState: defaultESState,
+			},
+			want: newResults(),
+			wantNewPartition: map[string]int32{
+				"default": 0,
+			},
+			wantSyncedFlush: true,
+		},
+		{
+			name: "just one (master) at a time",
+			args: args{
+				statefulSets: sset.StatefulSetList{
+					nodespec.TestSset{
+						Name:      "default",
+						Replicas:  3,
+						Master:    true,
+						Data:      true,
+						Partition: 3,
+						Status: appsv1.StatefulSetStatus{
+							CurrentRevision: "a",
+							UpdateRevision:  "b",
+						},
+					}.Build(),
+				},
+				esState: defaultESState,
+			},
+			want: newResults().WithResult(defaultRequeue),
+			wantNewPartition: map[string]int32{
+				"default": 2,
+			},
+			wantSyncedFlush: true,
+		},
+		{
+			name: "multiple ssets, update correct sset",
+			args: args{
+				statefulSets: sset.StatefulSetList{
+					nodespec.TestSset{
+						Name:     "master",
+						Replicas: 3,
+						Master:   true,
+					}.Build(),
+					nodespec.TestSset{
+						Name:      "data",
+						Replicas:  1,
+						Data:      true,
+						Partition: 1,
+						Status: appsv1.StatefulSetStatus{
+							CurrentRevision: "a",
+							UpdateRevision:  "b",
+						},
+					}.Build(),
+				},
+				esState: defaultESState,
+			},
+			want: newResults(),
+			wantNewPartition: map[string]int32{
+				"data": 0,
+			},
+			wantSyncedFlush: true,
+		},
+		{
+			name: "wait for healthy cluster",
+			args: args{
+				statefulSets: sset.StatefulSetList{
+					nodespec.TestSset{
+						Name:      "default",
+						Replicas:  1,
+						Master:    true,
+						Data:      true,
+						Partition: 2,
+						Status: appsv1.StatefulSetStatus{
+							CurrentRevision: "a",
+							UpdateRevision:  "b",
+						},
+					}.Build(),
+				},
+				esState: mockESState{
+					green: false,
+				},
+			},
+			want:             newResults().WithResult(defaultRequeue),
+			wantNewPartition: map[string]int32{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var runtimeObjects []runtime.Object
+			for i := range tt.args.statefulSets {
+				runtimeObjects = append(runtimeObjects, &tt.args.statefulSets[i])
+			}
+			k8sClient := k8s.WrapClient(fake.NewFakeClient(runtimeObjects...))
+			d := &defaultDriver{
+				DefaultDriverParameters: DefaultDriverParameters{
+					Client: k8sClient,
+					Scheme: scheme.Scheme,
+				},
+			}
+			mu := mockUpdater{}
+			fc := fakeESClient{}
+			if got := d.doRollingUpgrade(tt.args.statefulSets, &fc, tt.args.esState, mu.updatePartition); !reflect.DeepEqual(got, tt.want) {
```

Comment:
My first thought here was that this could be an `assert.DeepEqual()` call, but looks like we want the information provided by `Aggregate()` to be able to debug better?

Reply:
testify does not have an `assert.DeepEqual` afaik

Reply:
I got rid of the `Aggregate()`

Reply:
>testify does not have an assert.DeepEqual afaik

Ah you're right, I landed on the godoc of a different assert package 🥇 . Looks like assert.Equal ends up calling reflect.DeepEqual though